src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/VerificationComponents/VerifyGitopsEntities.tsx

```diff
@@ -0,0 +1,199 @@
+  const entityIds = useMemo((): { application: string; cluster: string; repo: string } => {
+    const data = stepsProgress?.[CDOnboardingSteps.WHAT_TO_DEPLOY]?.stepData as WhatToDeployType
+    const artifactId = data.artifactSubType ? data.artifactSubType?.id : (data.artifactType?.id as string)
+
+    return GITOPS_ENTITY_IDS_BY_DEPLOYMENT_TYPE[artifactId] || {}
+  }, [])
+  const {
+    loading: clusterLoading,
+    refetch: refetchCluster,
+    error: clusterFailed,
+    data: clusterData
+  } = useAgentClusterServiceGet({
+    agentIdentifier: configdata?.agentInfo?.identifier as string,
+    identifier: entityIds?.cluster,
+    queryParams: { accountIdentifier: accountId, orgIdentifier, projectIdentifier }
+  })
+  const {
+    loading: repoLoading,
+    refetch: refetchRepo,
+    error: repoFailed,
+    data: repoData
+  } = useAgentRepositoryServiceGet({
+    agentIdentifier: configdata?.agentInfo?.identifier as string,
+    identifier: entityIds?.repo,
+    queryParams: { accountIdentifier: accountId, orgIdentifier, projectIdentifier }
+  })
+  const {
+    loading: applicationLoading,
+    refetch: refetchApp,
+    error: appFailed,
+    data: appData
+  } = useAgentApplicationServiceGet({
+    queryName: entityIds?.application,
+    agentIdentifier: configdata?.agentInfo?.identifier as string,
+    queryParams: { accountIdentifier: accountId, orgIdentifier, projectIdentifier }
+  })
+
+  const setupState = React.useMemo((): PipelineSetupState => {
+    return stepsProgress?.[CDOnboardingSteps.DEPLOYMENT_STEPS]?.stepData
+  }, [stepsProgress])
+
+  useMemo((): string => {
+    const data = stepsProgress?.[CDOnboardingSteps.HOW_N_WHERE_TO_DEPLOY]?.stepData as WhereAndHowToDeployType
+    return data?.agentInfo?.identifier as string
+  }, [])
+
+  const successIconProps = React.useMemo((): { icon: IconName; iconProps: Omit<IconProps, 'name'> } => {
+    return {
+      icon: 'tick-circle',
+      iconProps: { padding: { right: 'medium' }, color: Color.GREEN_700 }
+    }
+  }, [])
+  const failIconProps = React.useMemo((): { icon: IconName; iconProps: Omit<IconProps, 'name'> } => {
+    return {
+      icon: 'danger-icon',
+      iconProps: { padding: { right: 'medium' }, color: Color.ERROR, className: css.redIcon }
+    }
+  }, [])
+
+  const loadingIconProps = React.useMemo((): { icon: IconName; iconProps: Omit<IconProps, 'name'> } => {
+    return {
+      icon: 'steps-spinner',
+      iconProps: { padding: { right: 'medium' } }
+    }
+  }, [])
+
+  useEffect(() => {
+    appData?.name &&
+      trackEvent(ONBOARDING_INTERACTIONS.APP_VERIFICATION_SUCCESS, getBranchingProps(stepsProgress, getString))
+    repoData?.repository &&
+      trackEvent(ONBOARDING_INTERACTIONS.REPO_VERIFICATION_SUCCESS, getBranchingProps(stepsProgress, getString))
+    clusterData?.cluster &&
+      trackEvent(ONBOARDING_INTERACTIONS.CLUSTER_VERIFICATION_SUCCESS, getBranchingProps(stepsProgress, getString))
```

Comment:
I would avoid these kind of short-hand `if`s. They're not immediately obvious and take a bit of reading to figure out what's going on.
```suggestion
    if (appData?.name ) {
      trackEvent(ONBOARDING_INTERACTIONS.APP_VERIFICATION_SUCCESS, getBranchingProps(stepsProgress, getString))
    }
    
    if (repoData?.repository) {
      trackEvent(ONBOARDING_INTERACTIONS.REPO_VERIFICATION_SUCCESS, getBranchingProps(stepsProgress, getString))
    }
    
    if (clusterData?.cluster) {
      trackEvent(ONBOARDING_INTERACTIONS.CLUSTER_VERIFICATION_SUCCESS, getBranchingProps(stepsProgress, getString))
    }
```